### PR TITLE
[FIX] stock: avoid MemError on `_get_orderpoint_action`

### PR DESCRIPTION
**Pull request by aj-fuentes**
When we have too many products from orderpoints we cannot compute the
forecasts for all at once. Otherwise we may get a MemoryError. Here we
iterate by chunks.

This error was observed on upg-417045, where there were 381927 products.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock/models/stock_orderpoint.py [hunk 1]
```diff
@@ -335,24 +335,22 @@ def _get_orderpoint_action(self):
...
-                    continue
-                product_loc_by_qty[(product, loc)] = quantity
-
-        # group product by lead_days and location in order to read virtual_available
-        # in batch
-        for (product, loc), qty in product_loc_by_qty.items():
-            rules = product._get_rules_from_location(loc)
-            lead_days = rules.with_context(bypass_delay_description=True)._get_lead_days(product)[0]
-            ploc_per_day[(lead_days, loc)].add(product.id)
+        for chunk in map(self.env['product.product'].browse, split_every(5000, all_product_ids)):
```
Comment (amoyaux) [inline addons/stock/models/stock_orderpoint.py:342]: You could use split_every

Comment (aj-fuentes) [reply addons/stock/models/stock_orderpoint.py:342]: I'm using `split_every`, did you mean to use in a different way? :)

Comment (amoyaux) [reply addons/stock/models/stock_orderpoint.py:342]: Oups my bad. Not it's okay this way

Comment (amoyaux) [reply addons/stock/models/stock_orderpoint.py:342]: Just maybe rename chunk in products, it would be easier to read

Comment (aj-fuentes) [reply addons/stock/models/stock_orderpoint.py]: Done

**Comment by amoyaux** [comment]
robodoo r+

**Pull request closed without merging**